### PR TITLE
Remove kir::Expr::scope_

### PR DESCRIPTION
**Pull request by csarofeen**
Dependent on: https://github.com/csarofeen/pytorch/pull/1339 to avoid conflicts.

### CODE REVIEW

File: torch/csrc/jit/ir/alias_analysis.h [hunk 1]
```diff
@@ -155,7 +155,7 @@ class AliasDb {
   // Copy `from`s aliasing info to `to`.
   void copyValue(Value* from, Value* to);
   // Create a new `value` that does not alias anything else.
-  void createValue(const Value* value);
+  TORCH_API void createValue(const Value* value);
```
Comment (naoyam) [inline torch/csrc/jit/ir/alias_analysis.h]: Is this related to this PR?

File: torch/csrc/jit/codegen/cuda/graph_fuser.cpp [hunk 2]
```diff
@@ -2042,6 +2042,9 @@ void replaceAliasOpsWithCopy(std::shared_ptr<Graph>& graph, Block* block) {
         graph->insertNode(graph->create(op_mapping[n->kind()], n->inputs(), 1));
     op_copy->output()->setType(n->output(0)->type());
 
+    // adding newly created value into alias_db;
+    alias_db->createValue(op_copy->output());
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/graph_fuser.cpp]: Is this related to this PR?

File: torch/csrc/jit/codegen/cuda/kernel_ir.h [hunk 3]
```diff
@@ -271,19 +271,6 @@ class TORCH_CUDA_CU_API Expr : public Node {
     return outputs_;
   }
 
-  // TODO: Remove, requires removing scope_utils::getLoops
-  Scope* scope() const {
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/kernel_ir.h:275]: Do we need to keep the `Scope` class itself?

**Review by naoyam — APPROVED** [review]
The changes around `kir::Scope` look good to me. I'm not really sure about the changes around aliasing.

File: torch/csrc/jit/codegen/cuda/kernel_ir.h (hunk 3 above)
Comment (naoyam) [reply torch/csrc/jit/codegen/cuda/kernel_ir.h:275]: Never mind, the class itself is still needed.

File: torch/csrc/jit/codegen/cuda/graph_fuser.cpp (hunk 2 above)
Comment (csarofeen) [reply torch/csrc/jit/codegen/cuda/graph_fuser.cpp]: No, looks like I picked up an extra commit somehow, must have been out of sync.
